GenderPayGap.WebUI/BackgroundJobs/ScheduledJobs/SendReminderEmailsJob.cs

```diff
@@ -88,43 +91,69 @@ private void SendReminderEmailsForSectorType(SectorTypes sector, string runId, D
                             break;
                         }
 
-                        CheckUserAndSendReminderEmailsForSectorTypeAndReportingYear(user, sector, year);
+                        ReminderEmail reminderEmail;
+
+                        try
+                        {
+                            reminderEmail = GetReminderEmailRecord(user, sector, latestReminderEmailDate);
+                        }
+                        catch (Exception ex)
+                        {
+                            CustomLogger.Information(
+                                "Failed whilst saving reminder email",
+                                new
+                                {
+                                    user.UserId,
+                                    SectorType = sector,
+                                    Exception = ex.Message
+                                });
+                            continue;
+                        }
+
+                        if (reminderEmail != null)
+                        {
+                            CheckUserAndSendReminderEmailsForSectorTypeAndReportingYear(
+                                user,
+                                year,
+                                reminderEmail);
+                        }
                     }
                 }
             }
         }
 
-        private void CheckUserAndSendReminderEmailsForSectorTypeAndReportingYear(User user, SectorTypes sector, int year)
+        private void CheckUserAndSendReminderEmailsForSectorTypeAndReportingYear(User user, int year, ReminderEmail reminderEmail)
```

Comment:
This function's name also says check user, but from a quick look it doesn't seem to check the user. Is the "SendReminderEmailsForSectorTypeAndReportingYear" function the one that checks the user, and thus should we update its name? Maybe the parent function itself shouldn't have CheckUser in the name, since that could be considered an implementation detail?

Reply:
Good point! I'll remove the "SectorType" from the function name. I _think_ the "Check" stands for checking if there are any employers to email about - if the user has no registered employers then no email will be sent.